pkg/gofr/datasource/mongo/errors.go

```diff
@@ -0,0 +1,17 @@
+package mongo
+
+import "errors"
+
+var (
+	// ErrInvalidURI is returned when the MongoDB URI is invalid or cannot be parsed.
+	ErrInvalidURI = errors.New("invalid MongoDB URI")
+
+	// ErrAuthentication is returned when authentication fails.
+	ErrAuthentication = errors.New("authentication failed")
+
+	// ErrDatabaseConnection is returned when the client fails to connect to the specified database.
+	ErrDatabaseConnection = errors.New("failed to connect to database")
+
+	// ErrGenericConnection is returned for general connection issues.
+	ErrGenericConnection = errors.New("MongoDB connection error")
```

Comment:
Here is a suggestion, feel free to consider it or ignore it


```suggestion
	// ErrInvalidURI is returned when the MongoDB URI is invalid or cannot be parsed.
	ErrInvalidURI = fmt.Errorf("%w: invalid MongoDB URI", ErrConnection)

	// ErrAuthentication is returned when authentication fails.
	ErrAuthentication = fmt.Errorf("%w: authentication failed", ErrConnection)

	// ErrDatabaseConnection is returned when the client fails to connect to the specified database.
	ErrDatabaseConnection = fmt.Errorf("%w: failed to connect to database", ErrConnection)

	// ErrGenericConnection is returned for general connection issues.
	ErrConnection = errors.New("MongoDB connection error")
```

This way the caller can catch all mongodb error with an errors.Is(err, ErrConnection)